### PR TITLE
feat(organizations): add tabbed dashboard with overview tab

### LINKED ISSUE #7567: feature: Organization Dashboard UI with member search, role filter & pagination
### Is your feature request related to a problem? Please describe

Currently, CircuitVerse has the backend model, policy, and routing layer for Organizations, but there is no dedicated dashboard UI for organization members to interact with. Admins and mentors have no visual way to browse their organization's groups, view members by role, or navigate to organization settings, making the feature practically unusable from the frontend

### Describe the solution you'd like

A fully redesigned **Organization Dashboard** (/organizations/:id) that provides:

- Tabbed navigation - Groups, Members, and Settings tabs on the same page
- Groups tab - Displays all groups belonging to the organization in a card grid (9 per page) with pagination
- Members tab - Displays all organization members in a list with:

1. Live search by name (debounced, no page reload required)
2. Role filter (Admin / Mentor / Member)
3. Role-ordered listing (Admins first, then Mentors, then Members)
4. Pagination (10 per page)

- Settings tab - Links to the existing edit page (/organizations/:id/edit) where admins can update org details, logo, links, and location
- Organization header - Displays the org logo, name, description, colored social link icons (GitHub, LinkedIn, Instagram, Discord, YouTube, etc.), and location

### Describe alternatives you've considered

_No response_

### Additional context

GSoC'26 Project - 6 Enterprise & Institutional Organization Features

### Are you working on this?

Yes

### PR DESCRIPTION
**Pull request by naman79820**
Fixes #7567

#### Describe the changes you have made in this PR -

This PR adds the foundation for the new tabbed organization dashboard.

- Replaces the old `show` page with a tabbed dashboard: a shared shell (organization name, tab navigation, and header with logo, description, social links, and location) plus the **Overview** tab showing the organization's groups in a grid.
- Adds `overview`, `members`, `settings`, and `check_slug` routes.
- `show` now redirects to the `overview` tab, so `/organizations/:id` still works.
- The **Members** and **Settings** tabs are stubbed in this PR (placeholder content) and are completed in stacked follow-up PRs, so this PR stays focused on the dashboard structure and overview.
- Adds the dashboard styling (`organizations.scss`).

This is the first of three stacked PRs (dashboard → members → settings). Reviewing/merging in that order is recommended.

### Screenshots of the UI changes (If any) -

https://github.com/user-attachments/assets/4cedfed2-b52a-42cf-a37d-7c350ff08c98

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

The old organization page was a single `show` view. I restructured it into a tabbed dashboard (Overview / Members / Settings) using path-based routes (`/organizations/:id/overview`, `/members`, `/settings`) rather than query params, so each tab is a real, shareable URL and works with the browser back/forward buttons.

All three tabs share one shell partial (`_dashboard_shell`) that renders the org name, tab bar, and header, then yields the tab-specific content below via `content_for`. This keeps the header consistent across tabs and avoids duplication. Each tab's controller action sets `@active_tab`, which the shell uses to highlight the current tab.

`show` redirects to `overview` so the base URL stays valid. `check_slug` is a small JSON endpoint used to preview the organization's URL slug. I considered keeping the tabs as query params on a single `show` action, but path-based routes are cleaner, more RESTful, and better for sharing/SEO.

To keep this PR focused and reviewable, I split the work into three stacked PRs — this one establishes the shell + overview + routes, and the Members and Settings tabs are added in follow-ups (their tabs are stubbed here so nothing 404s).

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [ ] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

## Summary by CodeRabbit

* **New Features**
  * Added organization dashboard UI with Overview and Members tabs, plus admin-only Settings.
  * Added groups listing with member counts, pagination, and admin-only “New Group” access.
  * Added organization header branding, descriptions, locations, and social/external icons, with logo fallbacks.
  * Added slug availability checking for organization URLs.
* **Bug Fixes**
  * Organization pages now always open on the Overview view.
  * Improved authorization safety for signed-out access and protected group mentor memberships.
  * Added robust organization link normalization/validation for valid http/https URLs. 
* **Tests**
  * Updated and added specs for redirects, policy edge cases, and link scheme validation.

### CODE REVIEW

File: app/views/organizations/_dashboard_shell.html.erb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                  <%= link_to link, target: "_blank", rel: "noopener noreferrer",
+                        class: "org-dash-social org-dash-social-x", title: link do %>
+                    <svg viewBox="0 0 24 24" width="25" height="25" fill="currentColor" aria-hidden="true">
+                      <path d="M18.244 2.25h3.308l-7.227 8.26 8.502 11.24H16.17l-5.214-6.817L4.99 21.75H1.68l7.73-8.835L1.254 2.25H8.08l4.713 6.231zm-1.161 17.52h1.833L7.084 4.126H5.117z"/>
+                    </svg>
+                  <% end %>
+                <% else %>
+                  <%= link_to link, target: "_blank", rel: "noopener noreferrer",
+                        class: "org-dash-social", title: link do %>
+                    <i class="org-social-icon <%= icon_for_url(link) %>" aria-hidden="true"></i>
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/views/organizations/_dashboard_shell.html.erb]: check it does consist now in this pr

File: app/controllers/organizations_controller.rb [hunk 2]
```diff
@@ -1,34 +1,62 @@
...
     @organization = Organization.new
   end
 
+  # GET /organizations/check_slug
+  def check_slug
+    base_slug = params[:slug].presence || params[:name].to_s.strip.parameterize
+    is_taken = base_slug.present? && Organization.exists?(slug: base_slug)
+
+    render json: { slug: base_slug, available: base_slug.present? && !is_taken }
+  end
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/controllers/organizations_controller.rb:64]: recheck

File: app/views/organizations/overview.html.erb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      <div class="org-dash-groups">
+        <% @groups.each do |group| %>
+          <%= link_to group_path(group), class: "org-dash-group-card" do %>
+            <span class="org-dash-group-name"><%= group.name %></span>
+            <span class="org-dash-group-meta">
+              <i class="fa fa-user" aria-hidden="true"></i>
+              <%= pluralize(group.group_members.size, "member") %>
+            </span>
+          <% end %>
+        <% end %>
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/views/organizations/overview.html.erb]: should be good now

[the remark this answers is not in the document]
Comment (naman79820) [reply app/views/organizations/overview.html.erb]: checked manually no more n+1 query now

File: app/views/organizations/_dashboard_shell.html.erb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        <div class="org-dash-info-meta">
+          <% org_links = (@organization.respond_to?(:links) && @organization.links.is_a?(Array)) ? @organization.links.reject(&:blank?) : [] %>
+          <% if org_links.any? %>
+            <div class="org-dash-socials">
+              <% org_links.each do |link| %>
+                <% if link =~ /(?:twitter\.com|x\.com)/i %>
+                  <%= link_to link, target: "_blank", rel: "noopener noreferrer",
+                        class: "org-dash-social org-dash-social-x", title: link do %>
+                    <svg viewBox="0 0 24 24" width="25" height="25" fill="currentColor" aria-hidden="true">
+                      <path d="M18.244 2.25h3.308l-7.227 8.26 8.502 11.24H16.17l-5.214-6.817L4.99 21.75H1.68l7.73-8.835L1.254 2.25H8.08l4.713 6.231zm-1.161 17.52h1.833L7.084 4.126H5.117z"/>
```
Comment (naman79820) [inline app/views/organizations/_dashboard_shell.html.erb]: recheck

File: app/views/organizations/_dashboard_shell.html.erb [hunk 5]
```diff
@@ -0,0 +1,70 @@
+<div class="org-dash">
+  <%# Organization name %>
+  <h1 class="org-dash-name"><%= @organization.name %></h1>
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/views/organizations/_dashboard_shell.html.erb]: looks good recheck

File: app/views/organizations/_dashboard_shell.html.erb [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+      <div class="org-dash-logo">
+        <% if @organization.logo.attached? %>
+          <%= image_tag @organization.logo, alt: "#{@organization.name} logo" %>
+        <% else %>
+          <span class="org-dash-logo-fallback"><%= @organization.name.first.upcase %></span>
+        <% end %>
+      </div>
+
+      <div class="org-dash-info">
+        <h2 class="org-dash-info-name"><%= @organization.name %></h2>
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/views/organizations/_dashboard_shell.html.erb]: looks good recheck

File: app/component/organization_dashboard_shell_component.html.erb [hunk 7]
```diff
@@ -0,0 +1,15 @@
+<div class="org-dash">
+  <h1 class="org-dash-name"><%= @organization.name %></h1>
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/component/organization_dashboard_shell_component.html.erb]: false

**Comment by naman79820** [comment]
Hey @coderabbitai GitHub/X/YouTube/Facebook don't use regional subdomains that users would paste but LinkedIn uses country subdomains so do you think we should  add regix for it or no?

File: app/controllers/organizations_controller.rb [hunk 8]
```diff
@@ -3,32 +3,61 @@
...
+
+  # GET /organizations/1/overview
+  def overview
+    @active_tab = "overview"
+    @groups = @organization.groups
+                           .left_joins(:group_members)
+                           .select("groups.*, COUNT(group_members.id) AS group_members_count")
+                           .group("groups.id")
+                           .order(created_at: :desc)
+                           .paginate(page: params[:groups_page], per_page: 9, total_entries: @organization.groups.count)
```
Comment (PRATHAM2002-DS) [inline app/controllers/organizations_controller.rb]: This triggers an N+1: each group card in the view calls `group.group_members.size`, which fires a separate `SELECT COUNT(*)` per card. With 9 groups per page, that's 9 extra queries on every page load.

```suggestion
    @groups = @organization.groups.includes(:group_members)
                           .order(created_at: :desc)
                           .paginate(page: params[:groups_page], per_page: 9)
```

Going ahead we should also consider adding a `group_members_count` counter cache on the `Group` model so you don't load the association at all.

File: app/controllers/organizations_controller.rb [hunk 9]
```diff
@@ -3,32 +3,61 @@
...
-      current_user.organizations.order(created_at: :desc).paginate(page: params[:page])
+      current_user.organizations.order(created_at: :desc).paginate(page: params[:page], per_page: 9)
     end
   end
 
-  # GET /organizations/1
-  def show; end
+  # GET /organizations/1  → redirect to overview tab
+  def show
+    redirect_to overview_organization_path(@organization)
```
Comment (PRATHAM2002-DS) [inline app/controllers/organizations_controller.rb:23]: This redirect works, but it creates a double hop: `create` and `update` both redirect to `@organization` (which hits `show`), then `show` redirects again to `overview`. That's two round-trips on every create/update. Did you weigh if this is the only way to go?

File: app/assets/stylesheets/organizations.scss [hunk 10]
```diff
@@ -0,0 +1,269 @@
+$org-green:      #42b983;
+$org-green-dark: #026e57;
```
Comment (PRATHAM2002-DS) [inline app/assets/stylesheets/organizations.scss]: Is this how its hardcoded in other places in codebase?

File: app/controllers/organizations_controller.rb [hunk 11]
```diff
@@ -3,32 +3,61 @@
...
-  before_action :check_edit_access, only: %i[edit update destroy]
+  before_action :set_organization, only: %i[show overview members settings update destroy]
+  before_action :check_show_access, only: %i[show overview members]
+  before_action :check_edit_access, only: %i[settings update destroy]
 
   # GET /organizations
   def index
     @organizations = if params[:explore].present?
-      Organization.where(private: false).order(created_at: :desc).paginate(page: params[:page])
+      Organization.where(private: false).order(created_at: :desc).paginate(page: params[:page], per_page: 9)
```
Comment (PRATHAM2002-DS) [inline app/controllers/organizations_controller.rb]: nit: 9 should be a constant, rather make it configurable instead of hardcoding

File: app/models/organization.rb [hunk 12]
```diff
@@ -13,11 +13,13 @@ class Organization < ApplicationRecord
   attr_accessor :remove_logo
 
   before_validation { logo.purge if remove_logo == "1" }
+  before_validation :sanitize_links
 
-  validates :name, presence: true, uniqueness: { case_sensitive: false }
+  validates :name, presence: true, uniqueness: { case_sensitive: false }, length: { minimum: 2, maximum: 50 }
```
Comment (PRATHAM2002-DS) [inline app/models/organization.rb:19]: nit: again here use constants and if needed make it configurable

File: app/assets/stylesheets/theme.scss [hunk 13]
```diff
@@ -19,7 +19,6 @@ $navbar-dark-grey: #374151;
 $footer-dark-grey: #1c1c1c;
 $mini-button-blue: #42b9b6;
 $form-border-grey: #f2f2f2;
-$border-light-grey: #e5e5e5;
```
Comment (naman79820) [inline app/assets/stylesheets/theme.scss:22]: duplicate scss (already define at line 12)

File: app/assets/stylesheets/organizations.scss (hunk 10 above)
Comment (naman79820) [reply app/assets/stylesheets/organizations.scss]: Removed all hardcoded scss

File: app/controllers/organizations_controller.rb (hunk 11 above)
Comment (naman79820) [reply app/controllers/organizations_controller.rb]: added constant for pagination and links (controller and models)

File: app/controllers/organizations_controller.rb (hunk 9 above)
Comment (naman79820) [reply app/controllers/organizations_controller.rb:23]: Now create and update directly redirect to overview tab instead of dashboard

File: app/views/organizations/overview.html.erb [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+      <% if @groups.blank? %>
+        <div class="org-dash-empty">
+          <i class="fa fa-layer-group" aria-hidden="true"></i>
+          <p><%= t("organizations.show.no_groups") %></p>
+        </div>
+      <% else %>
+        <div class="org-dash-groups">
+          <% @groups.each do |group| %>
+            <%= link_to group_path(group), class: "org-dash-group-card" do %>
+              <span class="org-dash-group-name"><%= group.name %></span>
```
Comment (PRATHAM2002-DS) [inline app/views/organizations/overview.html.erb]: nit: Check if you are handling for null values correctly

File: app/controllers/organizations_controller.rb [hunk 15]
```diff
@@ -3,38 +3,73 @@
...
   before_action :authenticate_user!
   before_action :check_organizations_feature_flag
-  before_action :set_organization, only: %i[show edit update destroy]
-  before_action :check_show_access, only: %i[show]
-  before_action :check_edit_access, only: %i[edit update destroy]
+  before_action :set_organization, only: %i[show overview members settings update destroy]
+  before_action :check_show_access, only: %i[show overview members]
+  before_action :check_edit_access, only: %i[settings update destroy]
+
+  PER_PAGE = 9
```
Comment (PRATHAM2002-DS) [inline app/controllers/organizations_controller.rb:10]: nit: See how these are made configurable in org, we can make these configurable in followup PR

File: app/views/organizations/overview.html.erb (hunk 14 above)
Comment (naman79820) [reply app/views/organizations/overview.html.erb]: ReConfirmed it can't be null

File: app/controllers/organizations_controller.rb (hunk 15 above)
Comment (naman79820) [reply app/controllers/organizations_controller.rb:10]: ReConfirmed it is declared constant eg - tags_controller.rb
